Source/Classes/Completer/BaseCompleter.cs

```diff
@@ -11,40 +11,71 @@
-                                                                     string parameterName,
-                                                                     string wordToComplete,
-                                                                     CommandAst commandAst,
-                                                                     IDictionary fakeBoundParameters);
-
-      protected static void SelectValues(string wordToComplete, IEnumerable<string> words, List<CompletionResult> values)
-      {
-         foreach (var word in words)
-         {
-            if (string.IsNullOrEmpty(wordToComplete) || word.StartsWith(wordToComplete, true, CultureInfo.InvariantCulture))
+    public abstract class BaseCompleter : IArgumentCompleter
+    {
+        protected readonly IPowerShell _powerShell;
+
+        /// <summary>
+        /// This constructor is used during unit testings
+        /// </summary>
+        /// <param name="powerShell">fake instance of IPowerShell used for testing</param>
+        protected BaseCompleter(IPowerShell powerShell) { this._powerShell = powerShell; }
+
+        /// <summary>
+        /// This constructor is used when running in a PowerShell session. It cannot be
+        /// loaded in a unit test.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        protected BaseCompleter() : this(new PowerShellWrapper(RunspaceMode.CurrentRunspace)) { }
+
+        public abstract IEnumerable<CompletionResult> CompleteArgument(string commandName,
+                                                                       string parameterName,
+                                                                       string wordToComplete,
+                                                                       CommandAst commandAst,
+                                                                       IDictionary fakeBoundParameters);
+
+        protected static void SelectValues(string wordToComplete, IEnumerable<string> words, List<CompletionResult> values)
+        {
+            foreach (var word in words)
+            {
+                if (string.IsNullOrEmpty(wordToComplete) || word.StartsWith(wordToComplete, true, CultureInfo.InvariantCulture))
+                {
+                    values.Add(GetCompletionResult(word));
+                }
+            }
+        }
+
+
+        private static CompletionResult GetCompletionResult(string word)
+        {
+            return GetCompletionResult(word,word,CompletionResultType.Text);
+        }
+
+        private static CompletionResult GetCompletionResult(string word, string wordLabel, CompletionResultType resultType)
+        {
+            // Only wrap in single quotes if they have a space. This makes it easier
+            // to use on macOs
+            return (new CompletionResult(word.Contains(" ") ? $"'{word}'" : word, wordLabel, resultType, wordLabel + ": " + word));
+
+        }
+
+        protected static void SelectValues(string wordToComplete,
```

Comment:
I basically added here a method with a different header to allow completer to have display names of completers to be different from values.